src/main/java/net/dancier/chatdancer/utils/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,42 @@
+package net.dancier.chatdancer.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Slf4j
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    // 404
+    @ResponseBody
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    @ExceptionHandler(NotFoundException.class)
+    public NotFoundException handleNotFound(NotFoundException ex) {
+        log.info("Requested Chat not found");
```

Comment:
You should not add a specififc log message here as every NotFoundException will be handled here (we might use the exception for additional cases). You should log the message from the exception that you handle here.